### PR TITLE
Enable components tests for splunk hex, sapm, signalfx exporters and receivers

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: receiver/sapmreceiver/trace_receiver.go [hunk 1]
```diff
@@ -44,9 +44,10 @@ var gzipWriterPool = &sync.Pool{
 // sapmReceiver receives spans in the Splunk SAPM format over HTTP
 type sapmReceiver struct {
 	settings component.TelemetrySettings
+	config   *Config
 
-	config *Config
-	server *http.Server
+	server     *http.Server
+	shutdownWG sync.WaitGroup
```
Comment (dmitryax) [inline receiver/sapmreceiver/trace_receiver.go:50]: Are these changes required for enabling receivers in the tests?

File: receiver/signalfxreceiver/factory.go [hunk 2]
```diff
@@ -108,7 +108,7 @@ func createMetricsReceiver(
 	}
 	receiverLock.Unlock()
 
-	r.RegisterMetricsConsumer(consumer)
+	r.registerMetricsConsumer(consumer)
```
Comment (dmitryax) [inline receiver/signalfxreceiver/factory.go:111]: Similar question here. Or it's unrelated?

File: receiver/sapmreceiver/trace_receiver.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/sapmreceiver/trace_receiver.go:50]: Correct, right now the component was incorrect because it did not wait for the server to close before exiting shutdown.

File: receiver/signalfxreceiver/factory.go (hunk 2 above)
Comment (bogdandrutu) [reply receiver/signalfxreceiver/factory.go:111]: Small nit here :D

File: receiver/sapmreceiver/trace_receiver.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/sapmreceiver/trace_receiver.go:50]: Without this the tests fail with binding already bounded port since server is not yet closed.

File: receiver/signalfxreceiver/factory.go (hunk 2 above)
Comment (dmitryax) [reply receiver/signalfxreceiver/factory.go:111]: You would ask to move this to another PR haha

File: receiver/sapmreceiver/trace_receiver.go (hunk 1 above)
Comment (dmitryax) [reply receiver/sapmreceiver/trace_receiver.go:50]: Ok, thanks for clarifying

**Review by dmitryax — APPROVED** [review]

File: receiver/signalfxreceiver/factory.go (hunk 2 above)
Comment (bogdandrutu) [reply receiver/signalfxreceiver/factory.go]: Point taken, removed 🗡️

**Review by dmitryax — APPROVED** [review]
